test/probe/http-client.test.js

```diff
@@ -82,9 +83,11 @@ function doAssertMetrics(metrics, opts) {
     assert.strictEqual(metrics.method, opts.method || "GET");
     assert.strictEqual(metrics.path, path);
     assert.strictEqual(metrics.url, opts.url || url);
-    assert.strictEqual(metrics.contentType, "application/json");
+    assert.strictEqual(metrics.contentType, opts.contentType || "application/json");
     if (!opts.ignoreServerIPAddress) {
-        assert.strictEqual(metrics.serverIPAddress, "127.0.0.1");
+        if (!["127.0.0.1","::1"].includes(metrics.serverIPAddress)) {
```

Comment:
Should we add `localhost` too? 
```suggestion
        if (!["127.0.0.1","::1"].includes(metrics.serverIPAddress)) {
```

Reply:
It's IP address. In node 16 or so it changed from reporting IPv4 to IPv6 for localhost loopbacks, hence we need to support both so it works in different node versions.